src/web/extractors.rs

```diff
@@ -425,6 +523,38 @@ mod tests {
         }
     }
 
+    fn create_valid_hawk_header(
+        payload: &HawkPayload,
+        state: &ServerState,
+        method: &str,
+        path: &str,
+        host: &str,
+        port: u16,
+    ) -> String {
+        let salt = payload.salt.clone();
+        let payload = serde_json::to_string(payload).unwrap();
+        let mut hmac: Hmac<Sha256> = Hmac::new_varkey(&state.secrets.signing_secret).unwrap();
+        hmac.input(payload.as_bytes());
+        let payload_hash = hmac.result().code();
+        let mut id = payload.as_bytes().to_vec();
+        id.extend(payload_hash.to_vec());
+        let id = base64::encode_config(&id, base64::URL_SAFE);
+        let token_secret = hkdf_expand_32(
+            format!("services.mozilla.com/tokenlib/v1/derive/{}", id).as_bytes(),
+            Some(salt.as_bytes()),
+            &SECRETS.master_secret,
+        );
```

Comment:
It feels like you should be able to import some higher-level function to do this, rather than duplicating all the details of how to make a token into the test here.

Reply:
Indeed, I didn't find any function that would do it though. I'll probably move this out to be that higher level function elsewhere.